luarules/gadgets/unit_attached_con_turret_mex.lua

```diff
@@ -0,0 +1,115 @@
+function gadget:GetInfo()
+    return {
+        name      = 'Legion Con Turret Metal Extractor',
+        desc      = 'Allows the mex to function as a con turret by replacing it with a fake mex with a con turret attached',
+        author    = 'EnderRobo',
+        version   = 'v1',
+        date      = 'September 2024',
+        license   = 'GNU GPL, v2 or later',
+        layer     = 12,
+        enabled   = true
+    }
+end
+
+if not gadgetHandler:IsSyncedCode() then
+    return false
+end
+
+local SpGetUnitPosition = Spring.GetUnitPosition
+local SpGetUnitDefID = Spring.GetUnitDefID
+local SpGetUnitAllyTeam = Spring.GetUnitAllyTeam
+local SpGetUnitHealth = Spring.GetUnitHealth
+local SpGetUnitBuildFacing = Spring.GetUnitBuildFacing
+local SpGetUnitIsDead = Spring.GetUnitIsDead
+local SpGetUnitStates = Spring.GetUnitStates
+
+function gadget:UnitFinished(unitID, unitDefID, unitTeam)
+
+	local unitDef = UnitDefs[unitDefID]
+	if unitDef.name == "legmohocon" then																-- checks for mex
+		local nano_id
+		local imex_id
+		local health
+		local facing
+		local xx,yy,zz
+		local bt,mc,ec
+		local em
+		xx,yy,zz = SpGetUnitPosition(unitID)
+		facing = SpGetUnitBuildFacing(unitID)
+		bt,mc,ec = Spring.GetUnitCosts(unitID)
+		health = SpGetUnitHealth(unitID)																-- saves location, rotation, cost and health of mex
+		imex_id = Spring.CreateUnit("legmohoconin",xx,yy,zz,facing,Spring.GetUnitTeam(unitID) )			-- creates imex on where mex was
+		Spring.UseTeamResource(unitTeam, "metal", mc)													-- creating imex reclaims mex, this removes the metal that would give. DestroyUnit doesnt prevent the reclaim
+		if not imex_id then
+			return
+		end
+		Spring.SetUnitBlocking(imex_id, true, false, false)												-- makes imex non interactive
+		Spring.SetUnitNoSelect(imex_id,true)
+		nano_id = Spring.CreateUnit("legmohoconct",xx,yy,zz,facing,Spring.GetUnitTeam(imex_id) )		-- creates con on imex
+		if not nano_id then
+			return
```

Comment:
failing and giving up half way through after already spawning the first half of the two unit mex doesn't seem like a good idea?
tho i can't recommend a better alternative.

Reply:
Don't think this will ever fire, but in case it does I added a script to remove the whole unit and refund it. I'm guessing it could fire if the player hit the unit cap.